docs/source/using/host-folders.adoc

```diff
@@ -0,0 +1,224 @@
+
+[[minishift-host-folders]]
+== Minishift hostfolder command
+
+Minishift provides the link:../command-ref/minishift_hostfolder{outfilesuffix}[`hostfolder`] command to list, add, mount, unmount and remove host folders.
+In contrast to the <<driver-host-folders, driver provided host folders>>, the `hostfolder` command allows you to mount multiple shared folders onto custom specified mount points.
+
+At the moment only https://en.wikipedia.org/wiki/Server_Message_Block[CIFS] is supported as host folder type.
+Support for https://en.wikipedia.org/wiki/SSHFS[SSHFS] based host folders is in progress (see GitHub issue link:https://github.com/minishift/minishift/issues/317[#317]).
+If you want to use SSHFS now, refer to <<sshfs-folder-mount>> for manual instructions.
+
+The following sections describe the prerequisite for using the `hostfolder` command, as well as the various sub-commands it provides.
+
+[[host-folder-prerequisite]]
+=== Prerequisite
+
+To use the hostfolder command, you need to be able to share directories using CIFS.
+On Windows CIFS is the default technology for sharing directories.
+For example, on Windows 10 the _C:\Users_ folder is shared by default and can be accessed by locally-authenticated users.
+
+It is also possible to use CIFS on OS X and Linux.
+On the former you can enable CIFS based shares under _System Preferences > Sharing_.
+Refer to link:https://support.apple.com/en-us/HT204445[How to connect with File Sharing on your Mac] for detailed setup instructions.
+
+On Linux follow your distribution specific instructions to install link:https://www.samba.org[Samba].
+
+[[displaying-host-folders]]
+=== Displaying host folders
+
+The link:../command-ref/minishift_hostfolder_list{outfilesuffix}[`hostfolder list`] command gives you an overview of the defined host folders, their name, mount point, remote path and whether they are currently mounted.
+An example output could look like:
+
+----
+$ minishift hostfolder list
+Name    Mountpoint        Remote path          Mounted
+mac     /mnt/sda1/mac     //192.168.1.82/MAC   N
+----
+
+In this example, there is a host folder with the name _mac_ which mounts _//192.168.1.82/MAC_ onto _/mnt/sda1/mac_ in the Minishift VM.
+The share is currently not mounted.
+
+NOTE: The remote path must be reachable from within the VM.
+In the example above _192.168.1.82_ is the IP of host within the LAN which is one options one can use.
+You can use `ifconfig` (or `Get-NetIPAddress | Format-Table` on Windows) to determine a routable IP address.
+
+[[adding-host-folders]]
+=== Adding host folders
+
+The link:../command-ref/minishift_hostfolder_add{outfilesuffix}[`hostfolder add`] command allows you to define a new host folder.
+This in an interactive process which queries the relevant details for a host folder based on CIFS.
+<<adding-cifs-hostfolder>> shows an example.
```

Comment:
The <<adding-cifs-hostfolder>> section below shows an example. 
The link here really seems unnecessary.